moduleframework/mtf_scheduler.py

```diff
@@ -0,0 +1,163 @@
+        # TODO
+        description="""Start your tests!, unknown param are forwarded to avocado \n  USE param like abc,tag TODO par """, formatter_class=argparse.RawTextHelpFormatter
+    )
+    parser.add_argument('--verbose', '-v', action='store_true', default=False)
+    parser.add_argument("--name", "-n", action="store", default="name not given", help='Name of module for testing')
+    parser.add_argument("--linter", "-l", action="store_true", default=False, help='Name of module for testing')
+    parser.add_argument("--setup", "-s", action="store_true", default=False, help='Setup')
+    parser.add_argument("--action", "-a", action="store", default='run', help='Action for avocado')
+    parser.add_argument("--module", "-m", action="store", default='docker', help='Module type, like: docker, nspawn or rpm')
+
+    #parser.add_argument('files', type=argparse.FileType('r'), nargs='+')
+
+    # load args from cli and show loaded options
+    args, unknown = parser.parse_known_args()
+
+    set_logging(level=logging.DEBUG if args.verbose else logging.INFO)
+
+    logger.debug("Options: verbose={0}, name={1}, linter={2}, setup={3}, action={4}, module={5}".format(
+        args.verbose, args.name, args.linter, args.setup, args.action, args.module))
+    logger.debug("remaining options for avocado: {0}".format(unknown))
+
+    # environment
+    # hint for read: os.environ.get('MODULE') ; write os.environ['MODULE']
+    if os.environ.get('MODULE') is None:
+        os.environ['MODULE'] = args.module
+        logger.debug("set up environment {}".format(os.environ['MODULE']))
+    logger.debug("MODULE={0}".format(os.environ.get('MODULE')))
+
+
+    return args, unknown
+
+class AvocadoStart(object):
+    def __init__(self, args, unknown):
+
+        self.module = os.environ['MODULE']
+        self.action = args.action
+
+        # its used for filepath in loadCli:
+        self.tmppath = tempfile.mktemp()
+        site_libs = os.path.dirname(moduleframework.__file__)
+        mtf_tools = "tools"
+
+        # choose between TESTS and ADDITIONAL ENVIRONMENT from options
+        self.tests = ''
+        if args.linter:
+            self.tests = ("{SITE_LIB}/{MTF_TOOLS}/*.py".format(SITE_LIB=site_libs, MTF_TOOLS=mtf_tools))
+        self.additionalAvocadoArg = ''
+
+        for param in unknown:
+            # take care of this, see tags for safe/unsafe:
+            # http://avocado-framework.readthedocs.io/en/52.0/WritingTests.html#categorizing-tests
+            if os.path.exists(param):
+                # this is test
+                self.tests += " {0} ".format(param)
+            else:
+                # this is additional avocado param
+                self.additionalAvocadoArg += " {0} ".format(param)
+
+        logger.debug("tests = {0}".format(self.tests))
+        logger.debug("additionalAvocadoArg = {0}".format(self.additionalAvocadoArg))
+
+
+    def bash_command(self, cmd):
+         return process.run(cmd, shell=True, ignore_status=True)
```

Comment:
We have self.runHost in MTF. I would prefer to remove it and makes it general. https://github.com/fedora-modularity/meta-test-family/blob/devel/moduleframework/common.py#L400

Reply:
self.runHost is not usable as it load config (which cannot exists) and after discussion with  @jscotka I should use own stuff in this case